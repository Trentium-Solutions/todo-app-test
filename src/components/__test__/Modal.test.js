import Modal from '../Modal'
import {createRoot} from 'react-dom/client'
import TestRenderer from 'react-test-renderer'
import {Provider} from 'react-redux'
import {PersistGate} from 'redux-persist/integration/react'
import {store, persistore} from '../../state/store'

describe('modal test suite', () => {
  const modalComponent = (
    <Provider store={store}>
      <PersistGate persistor={persistore}>
        <Modal />
      </PersistGate>
    </Provider>
  )
  it('renders without crashing', () => {
    const div = document.createElement('div')
    const root = createRoot(div)
    root.render(<Modal />)
  })

  it('matches snapshot modal', () => {
    const tree = TestRenderer.create(modalComponent).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
