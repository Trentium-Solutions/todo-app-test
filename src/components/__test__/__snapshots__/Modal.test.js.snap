// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal test suite matches snapshot modal 1`] = `null`;
